actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -435,7 +435,8 @@ def form_for(record, options = {}, &block)
           object      = nil
         else
           object      = record.is_a?(Array) ? record.last : record
-          raise ArgumentError, "First argument in form cannot contain nil or be empty" unless object
+          contains_nil = record.nil? || (record.is_a?(Array) && record.include?(nil))
```

Comment:
I think it's correct, it was `record.last` previously as well and if `[@article, @comment]` is passed, `@comment` is the object which the form should be generated for. 

Thanks for the warm welcome! 😄